main.go

```diff
@@ -100,44 +195,131 @@ func main() {
 			LAPIStream:      lapiStream,
 			UpdateFrequency: conf.CloudflareConfig.UpdateFrequency,
 			Wg:              &wg,
+			UpdatedState:    stateStream,
+			CFStateByAction: states,
+			Count:           Count,
+		}
+		if *onlySetup {
+			workerTomb.Go(func() error {
+				var err error = nil
+				defer func() {
+					workerTomb.Kill(err)
+					stateStream <- nil
+				}()
+
+				worker.CFStateByAction = nil
+				err = worker.Init()
+				if err != nil {
+					return err
+				}
+				err = worker.SetUpCloudflareIfNewState()
+				return err
+
+			})
+		} else if *delete {
+			workerTomb.Go(func() error {
+				var err error = nil
+				defer func() {
+					workerTomb.Kill(err)
+					stateStream <- nil
+				}()
+				err = worker.Init()
+				if err != nil {
+					return nil
+				}
+				err = worker.deleteExistingIPList()
+				return err
+
+			})
+		} else {
+			workerTomb.Go(func() error {
+				err := worker.Run()
+				return err
+			})
 		}
-		workerTomb.Go(func() error {
-			err := worker.Run()
-			return err
-		})
 	}
 	var dispatchTomb tomb.Tomb
+	var stateTomb tomb.Tomb
 
 	dispatchTomb.Go(func() error {
 		wg.Wait()
 		go csLapi.Run()
 		for {
-			select {
-			case decisions := <-csLapi.Stream:
-				// broadcast decision to each worker
-				for _, lapiStream := range lapiStreams {
-					lapiStream <- decisions
+			decisions := <-csLapi.Stream
+			// broadcast decision to each worker
+			for _, lapiStream := range lapiStreams {
+				lapiStream <- decisions
+			}
+		}
+	})
+
+	stateTomb.Go(func() error {
+		aliveWorkerCount := len(conf.CloudflareConfig.Accounts)
+		for {
+			newStates := <-stateStream
+			if newStates == nil {
+				aliveWorkerCount--
```

Comment:
newStates == nil only when the setup  of that worker is complete. It is used to  signal  this tomb that the worker is done with its setup. 

In other normal  cases we never receive nil state